src/test/java/com/vaadin/flow/ai/formfiller/utils/ComponentUtilsTest.java

```diff
@@ -0,0 +1,119 @@
+package com.vaadin.flow.ai.formfiller.utils;
+
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.checkbox.Checkbox;
+import com.vaadin.flow.component.checkbox.CheckboxGroup;
+import com.vaadin.flow.component.combobox.ComboBox;
+import com.vaadin.flow.component.combobox.MultiSelectComboBox;
+import com.vaadin.flow.component.datepicker.DatePicker;
+import com.vaadin.flow.component.datetimepicker.DateTimePicker;
+import com.vaadin.flow.component.grid.Grid;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.component.radiobutton.RadioButtonGroup;
+import com.vaadin.flow.component.textfield.*;
+import com.vaadin.flow.component.timepicker.TimePicker;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+public class ComponentUtilsTest {
+
+    @Test
+    public void testCreateMappingWorksForLayout() {
+        VerticalLayout formLayout = new VerticalLayout();
+        TextField textField = new TextField();
+        textField.setId("name");
+        formLayout.add(textField);
+        ComponentUtils.ComponentsMapping componentsMapping = ComponentUtils.createMapping(formLayout);
+        assertFalse(componentsMapping.components().isEmpty());
+        assertFalse(componentsMapping.componentsJSONMap().isEmpty());
+        assertFalse(componentsMapping.componentsTypesJSONMap().isEmpty());
+    }
+
+    @Test
+    @Ignore
```

Comment:
What prevent this test to pass?

Reply:
Thanks. I think this PR's changes should contain the fix for this test:
- https://github.com/vaadin/form-filler-addon/pull/98

I will investigate if it is fixed it or still failing with this change.